### PR TITLE
Added verification code for cassandra put operations as a config option.

### PR DESCRIPTION
**Pull request by dstuebe**
Pull request to add verification for all cassandra put operations.

### CODE REVIEW

File: ion/core/data/cassandra.py [hunk 1]
```diff
@@ -352,6 +355,16 @@ def put(self, key, value):
...
         yield self.client.batch_insert(key, self._cache_name, columns)
 
+        if cassandra_verify:
+            cv = yield self.has_key(key)
+            if cv is False:
+                raise CassandraError('Verification of put failed for key: %s' % binascii.b2a_hex(key))
+            elif cv is True:
+                log.info('Verified put key: %s' % binascii.b2a_hex(key))
+            else:
+                raise CassandraError('Unexpected condition in Cassandra Verify has key result! Key: %s' % binascii.b2a_hex(key))
```
Comment (daf) [inline ion/core/data/cassandra.py:365]: Recommend also printing the value of `cv`, so you also see what it is

File: ion/core/data/cassandra.py [hunk 2]
```diff
@@ -378,6 +391,24 @@ def batch_put(self, batch_request):
...
+
+            bad_keys = []
+            good_keys = 0
+            for key, res in cv.iteritems():
+                if res is False:
+                    bad_keys.append(key)
+                elif res is True:
+                    good_keys += 1
+                else:
+                    raise CassandraError('Unexpected condition in Cassandra Verify has key result! Key: %s' % binascii.b2a_hex(key))
```
Comment (daf) [inline ion/core/data/cassandra.py:405]: same comment

File: ion/core/data/cassandra.py [hunk 3]
```diff
@@ -557,6 +588,15 @@ def put(self, key, value, index_attributes=None):
...
         yield self.client.batch_insert(key, self._cache_name, index_cols)
 
+        if cassandra_verify:
+            cv = yield self.has_key(key)
+            if cv is False:
+                raise CassandraError('Verification of put failed for key: %s' % binascii.b2a_hex(key))
+            elif cv is True:
+                log.info('Verified put key: %s' % binascii.b2a_hex(key))
+            else:
+                raise CassandraError('Unexpected condition in Cassandra Verify has key result! Key: %s' % binascii.b2a_hex(key))
```
Comment (daf) [inline ion/core/data/cassandra.py:598]: etc

File: ion/core/data/cassandra.py [hunk 4]
```diff
@@ -585,6 +625,24 @@ def batch_put(self, batch_request):
...
+
+            bad_keys = []
+            good_keys = 0
+            for key, res in cv.iteritems():
+                if res is False:
+                    bad_keys.append(key)
+                elif res is True:
+                    good_keys += 1
+                else:
+                    raise CassandraError('Unexpected condition in Cassandra Verify has key result! Key: %s' % binascii.b2a_hex(key))
```
Comment (daf) [inline ion/core/data/cassandra.py:639]: more annoying emails!

**Comment by daf** [comment]
It looks like `batch_has_key` can return `None` as the value of any of the keys if the number of columns returned by its query is != 1.  This is probably a completely off the wall condition, and will likely be the only thing printed by my comments above, unless you expect something other than a T/F from the query.

So perhaps you can disregard the comments above.

**Comment by dstuebe** [comment]
Thanks Dave

**Pull request merged**